src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -190,6 +196,27 @@ private static bool IsValidMSBuildPath(string path)
 #endif
         }
 
+        private static bool CheckIfRunningTests(string processNameCommandLine, string processNameCurrentProcess)
+        {
+            // First check if we're running in a known test runner.
+            if (IsProcessInList(processNameCommandLine, s_testRunners) ||
+                IsProcessInList(processNameCurrentProcess, s_testRunners))
+            {
+#if FEATURE_APPDOMAIN
```

Comment:
Had to #if this out :(. I couldn't find an obvious way to do this so we'll revert to old behavior for .NET Core for now.